source/cpu/pstate.go

```diff
@@ -32,12 +35,68 @@ func detectPstate() (map[string]string, error) {
 		return nil, nil
 	}
 
-	// Only looking for turbo boost for now...
+	// Get global pstate status
+	data, err := ioutil.ReadFile(source.SysfsDir.Path("devices/system/cpu/intel_pstate/status"))
+	if err != nil {
+		return nil, fmt.Errorf("could not read pstate status: %s", err.Error())
+	}
+	status := strings.TrimSpace(string(data))
+
+	// Check that intel_pstate driver is being used by looping through policies
+	policies, err := ioutil.ReadDir(source.SysfsDir.Path("devices/system/cpu/cpufreq"))
+	if err != nil {
+		return nil, fmt.Errorf("cpufreq directory not available to check pstate: %s", err.Error())
+	}
+
+	features := map[string]string{}
+	for _, policy := range policies {
+		// Ensure at least one cpu is using this policy
+		cpus, err := ioutil.ReadFile(source.SysfsDir.Path("devices/system/cpu/cpufreq", policy.Name(), "affected_cpus"))
+		if err != nil || strings.TrimSpace(string(cpus)) == "" {
+			klog.Errorf("cpufreq policy %s has no affected cpus", policy.Name())
+			continue
+		}
+
+		// Check if policy is using intel_pstate scaling driver
+		driver, err := ioutil.ReadFile(source.SysfsDir.Path("devices/system/cpu/cpufreq", policy.Name(), "scaling_driver"))
+		if err != nil {
+			klog.Errorf("cpufreq scaling driver cannot be read: %s", err.Error())
+			continue
+		}
+
+		if strings.TrimSpace(string(driver)) != "intel_pstate" {
+			continue
+		}
```

Comment:
I'm not sure that we need/want to check this. Isn't the driver a global setting(?) That is, if `intel_pstate/status != off` then that implies that it is the acting scaling driver for all cpus. WDYT(?)

Reply:
Sounds good. Removed this check.